### DISCUSSION TITLE
Using "." within document IDs

### DISCUSSION THREAD

**Discussion by louisifn** [meilisearch/product#638] [Feedback & Feature Proposal]
I am using Meilisearch with Firestore and have a collection of documents with a web domain as a document ID.

By utilising a web domain as an ID, there is no duplicated data. However, by using a domain as a DocumentID I cannot use this collection with Meilisearch due to the restriction on utilising "." within the documentID.

Is anyone else struggling with this and is it something that can be amended within Meilisearch?

Many thanks!

**Comment by dureuill** [comment]
Hello @louisifn :wave:

For reference, the currently accepted format for document ID is described here: <https://docs.meilisearch.com/learn/core_concepts/primary_key.html#formatting-the-document-id>

The decision to restrict the document ID format pre-exists my joining the team, but I think the last time we discussed this, one reason was that we didn't want users to have to handle url-encoding the document id to [retrieve a document by id](https://docs.meilisearch.com/reference/api/documents.html#get-one-document). That reason doesn't explain why `.` is not allowed, though :grimacing:. 

Document ID format is something I'd personally like us to revisit at some point in the future, to capture using URIs as document ID, notably. In the meantime, a workaround I used when storing paths as document ID is to base64-encode them and store the base64 encoding instead of the path as document ID (you have to be careful to remove or replace some characters like `=` in the encoding though)

**Reply by louisifn** [reply]
That is brilliant - I have put that workaround into action. Thanks!

**Reply by dureuill** [reply]
Glad it helped 😊
